app/src/org/commcare/android/logging/ForceCloseLogger.java

```diff
@@ -112,12 +117,31 @@ public String getFilename() {
         }
 
         try {
-            HttpResponse response = generator.postData(URI, entity);
+            HttpResponse response = generator.postData(submissionUri, entity);
             ByteArrayOutputStream bos = new ByteArrayOutputStream();
             response.getEntity().writeTo(bos);
             Log.d(TAG, "Response: " + new String(bos.toByteArray()));
         } catch (IOException e) {
             e.printStackTrace();
+            return false;
+        }
+
+        return true;
+    }
+
+    private static void writeErrorToStorage(ForceCloseLogEntry entry) {
+        if (logStorage != null) {
+            logStorage.write(entry);
+        }
+    }
+
+    private static String getSubmissionUri() {
+        try {
+            SharedPreferences settings = CommCareApplication._().getCurrentApp().getAppPreferences();
+            return settings.getString(CommCarePreferences.PREFS_SUBMISSION_URL_KEY,
```

Comment:
There are a lot of places that have these two lines of code, might be good to eventually replace them with this method.